After a download failure, retries seem to get stuck

This happens when using `inflight .tmp`. 
```
# initial download attempt fails
2023-12-29 11:36:02,231 [INFO] sarracenia.flowcb.log after_accept accepted: (lag: 691.87 )  exchange: xs_COPERNICUS subtopic:  a file with baseUrl: https://zipper.dataspace.copernicus.eu/odata/v1/ relPath: S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE retrievePath: Products(3b64d29c-f379-4b3e-b80f-6a9250fce2d9)/$value
2023-12-29 11:39:27,750 [WARNING] sarracenia.transfer read_writelocal util/writelocal mismatched file length writing S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE.tmp. Message said to expect 1997301401 bytes.  Got 1050840239 bytes.
2023-12-29 11:39:27,751 [ERROR] sarracenia.flow download incomplete download only 1050840239 of expected 1997301401 bytes for S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE.tmp
2023-12-29 11:39:27,751 [INFO] sarracenia.flow do_download attempt 1 failed to download https://zipper.dataspace.copernicus.eu/odata/v1//S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE to /apps/sarra/public_data/20231229/COPERNICUS/Sentinel-1/GRD/2023/12/29/S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE

# the .tmp file is not deleted after the download failure. 

# retries fail because the file already exists and it won't try to overwrite it
2023-12-29 12:01:05,658 [INFO] sarracenia.flowcb.log after_accept accepted: (lag: 2195.29 )  exchange: xs_COPERNICUS subtopic:  a file with baseUrl: https://zipper.dataspace.copernicus.eu/odata/v1/ relPath: S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE retrievePath: Products(3b64d29c-f379-4b3e-b80f-6a9250fce2d9)/$value
2023-12-29 12:01:05,658 [WARNING] sarracenia.flow do_download inflight file already exists. race condition, deferring transfer of /apps/sarra/public_data/20231229/COPERNICUS/Sentinel-1/GRD/2023/12/29/S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE
2023-12-29 12:08:38,122 [INFO] sarracenia.flowcb.log after_accept accepted: (lag: 2647.76 )  exchange: xs_COPERNICUS subtopic:  a file with baseUrl: https://zipper.dataspace.copernicus.eu/odata/v1/ relPath: S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE retrievePath: Products(3b64d29c-f379-4b3e-b80f-6a9250fce2d9)/$value
2023-12-29 12:08:38,122 [WARNING] sarracenia.flow do_download inflight file already exists. race condition, deferring transfer of /apps/sarra/public_data/20231229/COPERNICUS/Sentinel-1/GRD/2023/12/29/S1A_IW_GRDH_1SDV_20231229T090354_20231229T090422_051867_064420_4E5A.SAFE
```

There's this note in the code:

https://github.com/MetPX/sarracenia/blob/62cfca0591b70aaac1585e01ecc8799bbe2fc2b7/sarracenia/flow/__init__.py#L1631

I think deleting the file if it's > 5 minutes old would be a good solution

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1923,6 +1923,8 @@ class Flow:
                         self.proto[self.scheme].update_file(new_inflight_path)
             elif len_written < 0:
                 logger.error("failed to download %s" % new_file)
+                if (self.o.inflight != None) and os.path.isfile(new_inflight_path):
+                    os.remove(new_inflight_path)
                 return False
             else:
                 if block_length == 0:
@@ -1944,6 +1946,8 @@ class Flow:
                             logger.error( f'download more {len_written} than expected {block_length} bytes for {new_inflight_path}' )
                         else:
                             logger.error( f'incomplete download only {len_written} of expected {block_length} bytes for {new_inflight_path}' )
+                        if (self.o.inflight != None) and os.path.isfile(new_inflight_path):
+                            os.remove(new_inflight_path)
                         return False
                 # when len_written is different than block_length
                 msg['size'] = len_written

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1627,11 +1627,19 @@ class Flow:
             # assert new_inflight_path is set.
 
             if os.path.exists(msg['new_inflight_path']):
+
                 if self.o.inflight:
+                    how_old = time.time() - os.path.getmtime(msg['new_inflight_path'])
                     #FIXME: if mtime > 5 minutes, perhaps rm it, and continue? what if transfer crashed?
-                    logger.warning(
-                        'inflight file already exists. race condition, deferring transfer of %s'
-                        % msg['new_path'])
+                    #       Added this with fixed value, should it be a setting?
+                    if how_old > 300:
+                        os.unlink( msg['new_inflight_path'] )
+                        logger.info(
+                            f"inflight file is {how_old}s old. Removed previous attempt {msg['new_path']}" )
+                    else:
+                        logger.warning(
+                            'inflight file already exists. race condition, deferring transfer of %s'
+                            % msg['new_path'])
                     self.worklist.failed.append(msg)
                     continue
                 # overwriting existing file.

